### PR TITLE
More generic `IntoIterator` trait instead of `Vec`

### PR DESCRIPTION
**Pull request by duchainer**

### CODE REVIEW

File: src/prepare.rs [hunk 1]
```diff
@@ -7,7 +7,11 @@ pub struct SqlWriter {
     pub string: String,
 }
 
-pub fn inject_parameters(sql: &str, params: Vec<Value>, query_builder: &dyn QueryBuilder) -> String {
+pub fn inject_parameters(
+    sql: &str,
+    params: Vec<Value>,
+    query_builder: &dyn QueryBuilder,
+) -> String {
```
Comment (Progdrasil) [inline src/prepare.rs]: Be careful with rustfmt, we dont want to polute they're commits

File: src/query/delete.rs [hunk 2]
```diff
@@ -225,10 +237,17 @@ impl DeleteStatement {
...
+    pub fn order_by_table_columns<T, C, I>(&mut self, cols: I) -> &mut Self
+    where
+        T: IntoIden,
+        C: IntoIden,
+        I: IntoIterator<Item = (T, C, Order)>,
+    {
+        self.order_by_columns(
+            cols.into_iter()
+                .map(|(t, c, o)| ((t.into_iden(), c.into_iden()), o))
+                .collect::<Vec<((Rc<dyn Iden>, Rc<dyn Iden>), Order)>>(),
```
Comment (Progdrasil) [inline src/query/delete.rs]: can simply be 
```suggestion
                .collect::<Vec<_>>(),
```

File: src/query/insert.rs [hunk 3]
```diff
@@ -79,8 +79,10 @@ impl InsertStatement {
     /// # Examples
     /// 
     /// See [`InsertStatement::values`]
-    pub fn columns<C>(&mut self, columns: Vec<C>) -> &mut Self
-        where C: IntoIden {
+    pub fn columns<C, I>(&mut self, columns: I) -> &mut Self
+    where C: IntoIden
+,I: IntoIterator<Item = C>
```
Comment (Progdrasil) [inline src/query/insert.rs]: formatting
```suggestion
    where 
        C: IntoIden,
        I: IntoIterator<Item = C>
```

File: src/query/delete.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+use std::rc::Rc;
+
```
Comment (Progdrasil) [inline src/query/delete.rs]: no longer needed
```suggestion
```

File: src/query/select.rs [hunk 5]
```diff
@@ -350,18 +354,33 @@ impl SelectStatement {
...
+    pub fn table_columns<T, C, I>(&mut self, cols: I) -> &mut Self
+    where
+        T: IntoIden,
+        C: IntoIden,
+        I: IntoIterator<Item = (T, C)>,
+    {
+        self.columns(
+            cols.into_iter()
+                .map(|(t, c)| (t.into_iden(), c.into_iden()))
+                .collect::<Vec<(Rc<dyn Iden>, Rc<dyn Iden>)>>(),
```
Comment (Progdrasil) [inline src/query/select.rs]:
```suggestion
                .collect::<Vec<_>>(),
```

File: src/query/select.rs [hunk 6]
```diff
@@ -863,18 +882,33 @@ impl SelectStatement {
...
+    pub fn group_by_table_columns<T, C,I>(&mut self, cols: I) -> &mut Self
+    where
+        T: IntoIden,
+        C: IntoIden,
+        I: IntoIterator<Item = (T, C)>,
+    {
+        self.group_by_columns(
+            cols.into_iter()
+                .map(|(t, c)| (t.into_iden(), c.into_iden()))
+                .collect::<Vec<(Rc<dyn Iden>, Rc<dyn Iden>)>>(),
```
Comment (Progdrasil) [inline src/query/select.rs]:
```suggestion
                .collect::<Vec<_>>(),
```

File: src/query/select.rs [hunk 7]
```diff
@@ -1139,10 +1183,17 @@ impl SelectStatement {
...
+    pub fn order_by_table_columns<T, C, I>(&mut self, cols: I) -> &mut Self
+    where
+        T: IntoIden,
+        C: IntoIden,
+        I: IntoIterator<Item = (T, C, Order)>,
+    {
+        self.order_by_columns(
+            cols.into_iter()
+                .map(|(t, c, o)| ((t.into_iden(), c.into_iden()), o))
+                .collect::<Vec<((Rc<dyn Iden>, Rc<dyn Iden>), Order)>>(),
```
Comment (Progdrasil) [inline src/query/select.rs]:
```suggestion
                .collect::<Vec<_>>(),
```

File: src/query/update.rs [hunk 8]
```diff
@@ -382,10 +397,17 @@ impl UpdateStatement {
...
+    pub fn order_by_table_columns<T, C, I>(&mut self, cols: I) -> &mut Self
+    where
+        T: IntoIden,
+        C: IntoIden,
+        I: IntoIterator<Item = (T, C, Order)>,
+    {
+        self.order_by_columns(
+            cols.into_iter()
+                .map(|(t, c, o)| ((t.into_iden(), c.into_iden()), o))
+                .collect::<Vec<((Rc<dyn Iden>, Rc<dyn Iden>), Order)>>(),
```
Comment (Progdrasil) [inline src/query/update.rs]:
```suggestion
                .collect::<Vec<_>>(),
```

File: src/query/update.rs [hunk 9]
```diff
@@ -1,3 +1,5 @@
+use std::rc::Rc;
+
```
Comment (Progdrasil) [inline src/query/update.rs]:
```suggestion
```

**Review by Progdrasil — COMMENTED** [review]
A few comments on formatting and turbofish

File: src/prepare.rs (hunk 1 above)
Comment (duchainer) [reply src/prepare.rs]: Fixed in  594b1f2 more-generic-into-iterator-instead-of-vec Remove accidental Rustfmt out of edited code

File: src/query/update.rs (hunk 8 above)
Comment (duchainer) [reply src/query/update.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/update.rs (hunk 9 above)
Comment (duchainer) [reply src/query/update.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/select.rs (hunk 7 above)
Comment (duchainer) [reply src/query/select.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/select.rs (hunk 6 above)
Comment (duchainer) [reply src/query/select.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/select.rs (hunk 5 above)
Comment (duchainer) [reply src/query/select.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/delete.rs (hunk 4 above)
Comment (duchainer) [reply src/query/delete.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/delete.rs (hunk 2 above)
Comment (duchainer) [reply src/query/delete.rs]: Fixed in eb83cfe more-generic-into-iterator-instead-of-vec Remove useless precision from Vec turbofish

File: src/query/insert.rs (hunk 3 above)
Comment (duchainer) [reply src/query/insert.rs]: Fixed in 4c5d358 more-generic-into-iterator-instead-of-vec Apply formatting on InsertStatement::columns()

**Pull request merged**